src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/ViewModels/Tournaments/TournamentTeamsListViewModel.cs

```diff
@@ -27,10 +27,14 @@ public TournamentTeamsListViewModel()
         /// </summary>
         /// <param name="source">All needed teams</param>
         /// <param name="tournamentId">Tournament id</param>
-        public TournamentTeamsListViewModel(List<TeamTournamentDto> source, int tournamentId)
+        public TournamentTeamsListViewModel(IEnumerable<TeamTournamentDto> source, ICollection<Division> divisions, int tournamentId)
         {
             TournamentId = tournamentId;
-            TeamsList = source.Select(TeamNameViewModel.Map).ToList();
+            TeamsList = source
+                        .Select(TeamNameViewModel.Map)
+                        .OrderBy(model => model.DivisionName)
+                        .ThenBy(model => TeamNameViewModel.GetGroupName(model, divisions))
```

Comment:
You can group by Id. But display Name

Reply:
I and you decided to group by group name on last Thursday